reorganized tree to match vscode (#7)

adietish opened the pull request:
fixes #7 

impl in this branch/PR:
![image](https://user-images.githubusercontent.com/25126/72984010-f2955e80-3de2-11ea-9350-1c8221634d0c.png)

current impl in vscode:
![image](https://user-images.githubusercontent.com/25126/72897837-25781d80-3d23-11ea-9b30-f2d512ddab12.png)

adietish commented:
Turning this draft to a PR. 
next step: impl' lifecycle to allow choosing/changing a current namespace.

adietish commented:
@jeffmaury @sbouchet please review

sbouchet commented on src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/AbstractResourcesProvider.kt:
```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
```
please change it for 2020

sbouchet commented on src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/NamespacesProvider.kt:
```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
```
please change it to 2020

sbouchet commented on src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/NamespaceProvider.kt:
```diff
@@ -37,7 +37,7 @@ class NamespaceProvider(
         val provider = kindProviders[resource::class.java]
         var added = false
```
why the logic is commented ? it will always return false from now ?

adietish replied:
good catch, missed this (generics are different in kotlin, so change wasn't obvious)

sbouchet commented on src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/NamespaceProvider.kt:
```diff
@@ -46,7 +46,7 @@ class NamespaceProvider(
         var removed = false
```
same as above .

adietish commented:
> as a general comment the new files should use 2020 copyright, and maybe all the modified classes should also indicates the 2020 year ( but it's a little bit extreme ).
> The kotlin language is very nice to read, but i also request @jeffmaury to review the PR.

Thanks for reviewing. Addressed your concerns.

adietish commented:
@jeffmaury dare to please review so that I can merge this?

jeffmaury commented:
Tried on my local Minishift/CDK cluster and got this exception:

```
io.fabric8.kubernetes.client.KubernetesClientException: namespaces is forbidden: User "developer" cannot watch namespaces at the cluster scope: no RBAC policy matched
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onFailure(WatchConnectionManager.java:206)
	at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:571)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:198)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	Suppressed: java.lang.Throwable: waiting here
		at io.fabric8.kubernetes.client.utils.Utils.waitUntilReady(Utils.java:135)
		at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.waitUntilReady(WatchConnectionManager.java:354)
		at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:759)
		at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:738)
		at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:69)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceWatch.watch(KubernetesResourceWatch.kt:47)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceWatch.add(KubernetesResourceWatch.kt:38)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceWatch.addAll(KubernetesResourceWatch.kt:34)
		at org.jboss.tools.intellij.kubernetes.model.Cluster.createResourceWatch(Cluster.kt:145)
		at org.jboss.tools.intellij.kubernetes.model.Cluster.watch(Cluster.kt:47)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceModel.createCluster(KubernetesResourceModel.kt:40)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceModel.<init>(KubernetesResourceModel.kt:33)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceModel.<init>(KubernetesResourceModel.kt:30)
		at org.jboss.tools.intellij.kubernetes.model.KubernetesResourceModel.<init>(KubernetesResourceModel.kt)
		at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
		at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
		at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
		at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
		at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
		at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:85)
		at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:62)
		at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:45)
		at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:209)
		at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:246)
		at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:213)
		at com.intellij.openapi.components.ServiceManager.doGetService(ServiceManager.java:52)
		at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:30)
		at org.jboss.tools.intellij.kubernetes.KubernetesToolWindowFactory.createToolWindowContent(KubernetesToolWindowFactory.kt:34)
		at com.intellij.openapi.wm.impl.ToolWindowImpl.ensureContentInitialized(ToolWindowImpl.java:526)
		at com.intellij.openapi.wm.impl.ToolWindowImpl.getContentManager(ToolWindowImpl.java:354)
		at com.intellij.openapi.wm.impl.commands.RequestFocusInToolWindowCmd.processRequestFocus(RequestFocusInToolWindowCmd.java:50)
		at com.intellij.openapi.wm.impl.commands.RequestFocusInToolWindowCmd.run(RequestFocusInToolWindowCmd.java:42)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.ToolWindowsPane$AddDockedComponentCmd.run(ToolWindowsPane.java:732)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.commands.FinalizableCommand.finish(FinalizableCommand.java:36)
		at com.intellij.openapi.wm.impl.commands.ApplyWindowInfoCmd.run(ApplyWindowInfoCmd.java:50)
		at com.intellij.openapi.wm.impl.CommandProcessor.runNext(CommandProcessor.java:78)
		at com.intellij.openapi.wm.impl.CommandProcessor.run(CommandProcessor.java:59)
		at com.intellij.openapi.wm.impl.CommandProcessor.execute(CommandProcessor.java:50)
		at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.execute(ToolWindowManagerImpl.java:596)
		at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.execute(ToolWindowManagerImpl.java:576)
		at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.activateToolWindow(ToolWindowManagerImpl.java:667)
		at com.intellij.openapi.wm.impl.ToolWindowManagerImpl$MyInternalDecoratorListener.activated(ToolWindowManagerImpl.java:2222)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:133)
		at com.intellij.util.EventDispatcher.access$100(EventDispatcher.java:25)
		at com.intellij.util.EventDispatcher$3.invoke(EventDispatcher.java:96)
		at com.intellij.openapi.wm.impl.$Proxy117.activated(Unknown Source)
		at com.intellij.openapi.wm.impl.InternalDecorator.fireActivated(InternalDecorator.java:239)
		at com.intellij.openapi.wm.impl.StripeButton.actionPerformed(StripeButton.java:287)
		at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
		at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
		at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
		at javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:308)
		at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
		at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
		at java.awt.Component.processMouseEvent(Component.java:6548)
		at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
		at com.intellij.openapi.wm.impl.StripeButton.processMouseEvent(StripeButton.java:278)
		at java.awt.Component.processEvent(Component.java:6313)
		at java.awt.Container.processEvent(Container.java:2237)
		at java.awt.Component.dispatchEventImpl(Component.java:4903)
		at java.awt.Container.dispatchEventImpl(Container.java:2295)
		at java.awt.Component.dispatchEvent(Component.java:4725)
		at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
		at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4526)
		at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
		at java.awt.Container.dispatchEventImpl(Container.java:2281)
		at java.awt.Window.dispatchEventImpl(Window.java:2746)
		at java.awt.Component.dispatchEvent(Component.java:4725)
		at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
		at java.awt.EventQueue.access$500(EventQueue.java:98)
		at java.awt.EventQueue$3.run(EventQueue.java:715)
		at java.awt.EventQueue$3.run(EventQueue.java:709)
		at java.security.AccessController.doPrivileged(Native Method)
		at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
		at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
		at java.awt.EventQueue$4.run(EventQueue.java:737)
		at java.awt.EventQueue$4.run(EventQueue.java:735)
		at java.security.AccessController.doPrivileged(Native Method)
		at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
		at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
		at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
		at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:664)
		at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
		at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
		at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
		at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
		at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
		at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
		at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Also the UI displays:

![image](https://user-images.githubusercontent.com/695993/73249864-057aab00-41b6-11ea-9db6-3624f11d33d1.png)

adietish commented:
@jeffmaury using an openshift cluster is not covered yet. That's for a different issue. The aim here is to simply match the tree layout as seen in our plugin for vscode.

adietish commented:
@jeffmaury can you please re-review this PR?
